### ISSUE TITLE
Pasting from a register inserts it to a new line

### ISSUE THREAD

**Issue by Dhq** [syl20bnr/spacemacs#9453]
I am having an annoying error where when I yank or delete a text object (for example a word) from one line, it automatically seems to include the line break. If I have yanked a word, I almost never want it to include the line break. See gif for a visual explanation, where I do a "yw" and try to paste it on the line above:

![spacemacs](https://user-images.githubusercontent.com/1615161/29400379-772491f2-832d-11e7-9e4e-eb8a6c3d2170.gif)

I am using Spacemacs Release 0.200.x
Thanks in advance!

**Comment by duianto** [comment]
Which mode (file extension) are you in?
Could you provide detailed reproduction steps, how are you pasting?
It's also helpful to include your system info. Press `SPC h d s`, to copy your info to the clipboard.

**Comment by Dhq** [comment]
Thanks for your answer!

I am in a .js file.

You can see it in the gif I posted. I basically to "yw" on a word, then "k" to go up one line and "p" to paste it. Results with the pasted word on a new line.

System info output (Same result on my mac though):

`#### System Info :computer:
- OS: windows-nt
- Emacs: 25.1.1
- Spacemacs: 0.200.7
- Spacemacs branch: master (rev. c2774bc)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(markdown html javascript helm spacemacs-prettier auto-completion emacs-lisp git github csharp markdown syntax-checking elm sql react org colors
          (shell :variables shell-default-height 30 shell-default-position 'bottom)
          version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS
`

**Comment by duianto** [comment]
It seems to be working as expected for me, on both the latest master and develop branches.

I added the javascript layer and restarted Emacs. Then I opened a new buffer, and typed the text from the gif, the buffer was then saved as a `.js`.

I followed the steps, except that I moved the cursor to the left on the first line (as demonstrated in the gif), otherwise it gets inserted between the "S" and "tartPage", this was the result:
```js
import ArticlePage StartPage from "./pages/StartPage";
import ArticlePage from "./pages/ArticlePage";
```

I noticed that your using Emacs 25.1.1, the latest is 25.2.2.
And Spacemacs 0.200.7 was released in January, the latest version is 0.200.9, it was released in April.

You could try to update them. If you want to be able to return to your current setup easily, then I suggest that you first make a backup, of the `.emacs.d` (or `.spacemacs.d` if you used that name) folder and the `.spacemacs` file.

#### System Info :computer:
- OS: windows-nt
- Emacs: 25.2.2
- Spacemacs: 0.200.9
- Spacemacs branch: master (rev. 8e1af145) and develop (rev. 2bd08116)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey
 (auto-completion :variables auto-completion-enable-help-tooltip 'manual)
 colors emacs-lisp git haskell helm html javascript markdown org spell-checking version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES

**Comment by Dhq** [comment]
It looks like an update fixed this issue! Should of course have thought of that. Big thanks for the help @duianto !
